### PR TITLE
changes based on feedback

### PR DESCRIPTION
**Pull request by davidbernick**
@MKonopko can you invite Martin and Daniel (I don't have their github names) to look at these changes? This is based on the feedback.

### CODE REVIEW

**Comment by MKonopko** [comment]
Added Dylan as a reviewer.  I've sent a request to Martin to join the ga4gh space.  I can add him after that.  He is @martin-kuba

File: AAI/AAIConnectProfile.md [hunk 1]
```diff
@@ -305,7 +303,7 @@ Payload:
 -   iss: MUST be able to be appended with .well-known/openid-configuration to
     get spec of broker.
 
--   sub: authenticated user unique identifier
+-   sub: authenticated user unique identifier. email is suggested but another identifier MAY be used for privacy reasons.
```
Comment (martin-kuba) [inline AAI/AAIConnectProfile.md]: I think that email should not be recommended as an unique identifier. It has multiple issues. Users usually use more than one email address. They change emails when they change their names, e.g. when they marry. Email is considered personal data by European GDPR (General Data Protection Regulation), it should not be disclosed without written consent and the user may require delete of it from all systems. 

You may want to google up the long discussions about eduPersonPrincipalName, eduPersonTargetedId, eduPersonUniqueId and samlSubjectID or read the "Motivation" part of http://docs.oasis-open.org/security/saml-subject-id-attr/v1.0/csprd03/saml-subject-id-attr-v1.0-csprd03.pdf

Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: So you don't think it's even worth the "suggestion" that email is to be used? That's reasonable. It's PII here in the States and it'll be part of whatever GDPR-like legislation eventually happens. It's already part of NIST-800-53 Appendix j (which is similar to GDPR). 

That said, what do YOU recommend we use for Sub as there will be no central authority handing out IDs.

Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: I know ELIXIR has an obfuscated sub but it's centralized. In a non-centralized approach, as we have, we need to have "SOME" sub. Maybe we just accept that 'subs' can clash as a risk and advise that implementers make their own risk acceptance to trust which Identity Providers/Brokers they allow access_tokens from? But that seems that it's maybe not a great solution...

Comment (martin-kuba) [reply AAI/AAIConnectProfile.md]: My suggestion is to take the ELIXIR approach, it means a randomly generated string with scope ("@somedomain.com") at the end.

File: AAI/AAIConnectProfile.md [hunk 2]
```diff
@@ -303,7 +303,7 @@ Payload:
 -   iss: MUST be able to be appended with .well-known/openid-configuration to
     get spec of broker.
 
--   sub: authenticated user unique identifier. email is suggested but another identifier MAY be used for privacy reasons.
+-   sub: authenticated user unique identifier. A broker MAY abstract the suggested user email address with a unique identifier provided it maintains a way to map the user. See the [ELIXIR](https://docs.google.com/document/d/1vOyW4dLVozy7oQvINYxHheVaLvwNsvvghbiKTLg7RbY) for an implementation example. The sub in this case is a unique identifier issued by ELIXIR that abstracts the user's "real" email. Downstream Clearinghouses will need to know how to handle the `sub` attribute.
```
[the remark this answers is not in the document]
Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: @martin-kuba have you seen the more recent change i made?

File: AAI/AAIConnectProfile.md [hunk 3]
```diff
@@ -323,15 +321,15 @@ Payload:
     “ga4gh” is the [scope for RI
     claims](https://docs.google.com/document/d/11Wg-uL75ypU5eNu2p_xh9gspmbGtmLzmdq5VfPHBirE/edit#bookmark=id.6jhrok8dem8m)).
 
--   ga4gh_userinfo_claims: A list of OIDC claim names that are present from the /userinfo endpoint that are incomplete in <ga4gh-spec-claims> that are attached to this access token. For complex OIDC claims with substructure, a dot-notation MAY be used to more precisely indicate which sub-claims contain more information within the /userinfo endpoint. Non-normative examples include:
+-   ga4gh_userinfo_claims: Required but MAY be empty. A list of OIDC claim names that are present from the /userinfo endpoint that are incomplete in <ga4gh-spec-claims> that MAY be attached to this access token. For complex OIDC claims with substructure, a dot-notation MAY be used to more precisely indicate which sub-claims contain more information within the /userinfo endpoint. Non-normative examples include:
```
Comment (davidbernick) [inline AAI/AAIConnectProfile.md]: @cdvoisin what language should be here?

Comment (cdvoisin) [reply AAI/AAIConnectProfile.md]: Required but MAY be an empty array. A list of GA4GH claim names that are available from the OIDC /userinfo endpoint as per the [User Info JWT Format](#claims-sent-to-data-holder-by-a-broker-via-userinfo) section of the specification. For complex GA4GH claims with substructure,  a dot-notation MAY be used to more precisely indicate which sub-claims contain content within the /userinfo endpoint. Non-normative examples include:

File: AAI/AAIConnectProfile.md [hunk 4]
```diff
@@ -323,15 +321,15 @@ Payload:
...
 [“ga4gh.ControlledAccessGrants”, “ga4gh.AffiliationAndRoles”] : indicates that only those two specific RI claims that exist within the “ga4gh” claim object would have additional content not included within the access token.
 
 -   `<ga4gh-spec-claims>`: Claims included as part of a GA4GH standard specification based on the scopes provided. This content MAY be incomplete (i.e. a subset of data elements) and more may be fetched as indicated by ga4gh_userinfo_claims. A non-normative example of `<ga4gh-spec-claims>` is: "ga4gh": {[ga4gh claims](https://docs.google.com/document/d/11Wg-uL75ypU5eNu2p_xh9gspmbGtmLzmdq5VfPHBirE)}
 
 #### Claims sent to Data Holder by a Broker via /userinfo
 
-Only the GA4GH claims truly must be as proscribed here. Refer to OIDC Spec for more information.
+Only the GA4GH claims truly must be as proscribed here. Refer to OIDC Spec for more information. The /userinfo endpoint MAY use `application/json`, but `application/jwt` is preferred and `application/jwt` MUST be signed. [UserInfo](https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse)
```
Comment (cdvoisin) [inline AAI/AAIConnectProfile.md]: "MUST be signed as per.." [UserInfo](...)

File: AAI/AAIConnectProfile.md [hunk 5]
```diff
@@ -189,18 +189,17 @@ Note: the above diagram shows how claims flow from a Claim Source (e.g. database
...
-            format](#ga4gh-jwt-format)
+        1.  This document makes no specifications for id_tokens.
 
-    2.  Access_tokens MUST be in JWT
+    2.  Access_tokens MUST be in JWS format  
 
         1.  Access tokens for GA4GH use MUST be in [this
             format](#ga4gh-jwt-format).
 
         2.  MAY have a limited set of claims with a larger list of claims
```
[the remark this answers is not in the document]
Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: Can you explain this one?

File: AAI/AAIConnectProfile.md [hunk 6]
```diff
@@ -189,18 +189,17 @@ Note: the above diagram shows how claims flow from a Claim Source (e.g. database
...
             accessed in /userinfo
 
-        3.  Broker SHOULD include a “ga4gh_userinfo_claims” claim as an array of
+        3.  Broker MAY include a “ga4gh_userinfo_claims” claim as an array of
```
[the remark this answers is not in the document]
Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: Done. Will commit.

File: AAI/AAIConnectProfile.md [hunk 7]
```diff
@@ -323,15 +321,15 @@ Payload:
...
 [“ga4gh”] : indicates that some RI claims are available beyond what is included in the access token but does not indicate which ones.
 [“ga4gh.ControlledAccessGrants”, “ga4gh.AffiliationAndRoles”] : indicates that only those two specific RI claims that exist within the “ga4gh” claim object would have additional content not included within the access token.
 
 -   `<ga4gh-spec-claims>`: Claims included as part of a GA4GH standard specification based on the scopes provided. This content MAY be incomplete (i.e. a subset of data elements) and more may be fetched as indicated by ga4gh_userinfo_claims. A non-normative example of `<ga4gh-spec-claims>` is: "ga4gh": {[ga4gh claims](https://docs.google.com/document/d/11Wg-uL75ypU5eNu2p_xh9gspmbGtmLzmdq5VfPHBirE)}
```
[the remark this answers is not in the document]
Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: move which part?

File: AAI/AAIConnectProfile.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (cdvoisin) [reply AAI/AAIConnectProfile.md]: This is if we want to remove access tokens from having partial claims and always use the /userinfo to get all claims. If Martin and others are okay with leaving it in (it is an optional, MAY after all), then perhaps elaborate on what a "limited set of claims" means:

Optionally MAY include a limited set of claims, if the overall token size does not exceed 2 kb, with a larger list of claims available via the /userinfo endpoint.

However, if you do remove claims from the access token, please update the access token format section to remove it from there as well. I'll defer to you and Martin for which path to take forward.

[the remark this answers is not in the document]
Comment (davidbernick) [reply AAI/AAIConnectProfile.md]: *I* think leaving it in as a MAY is fine.

**Pull request closed without merging**